src/main/java/com/dnastack/wes/cromwell/CromwellService.java

```diff
@@ -448,6 +523,63 @@ private JsonNode extractJsonNode(String value) throws IOException {
         }
     }
 
+    private void extractSecondaryLogFiles(Set<String> secondaryFileSet, Set<String> logFileSet, CromwellMetadataResponse metadataResponse){
+        Map<String, Object> outputs = metadataResponse.getOutputs();
+        if (outputs != null && !outputs.isEmpty()) {
+            outputs.values().forEach(output -> extractFilesFromValue(secondaryFileSet, output));
+        }
+        Map<String, List<CromwellTaskCall>> calls = metadataResponse.getCalls();
+        if (calls != null && !calls.isEmpty()) {
+            calls.values().stream().flatMap(List::stream).forEach(call -> extractSecondaryLogFilesFromCall(secondaryFileSet, logFileSet, call));
+        }
+    }
+
+    private void extractSecondaryLogFilesFromCall(Set<String> secondaryFileSet, Set<String> logFileSet, CromwellTaskCall call){
+        Map<String, Object> outputs = call.getOutputs();
+        if (outputs != null && !outputs.isEmpty()) {
+            outputs.values().forEach(output -> extractFilesFromValue(secondaryFileSet, output));
+        }
+        String stderr = call.getStderr();
+        String stdout = call.getStdout();
+        if (stderr != null && storageClient.isFile(stderr)) {
+            logFileSet.add(stderr);
+        }
+        if (stdout != null && storageClient.isFile(stdout)) {
+            logFileSet.add(stdout);
+        }
+        Map<String, String> backendLogs = call.getBackendLogs();
+        if (backendLogs != null && !backendLogs.isEmpty()) {
+            backendLogs.values().forEach(log -> extractFilesFromValue(logFileSet, log));
+        }
+        CromwellMetadataResponse subWorkflowMetadata = call.getSubWorkflowMetadata();
+        if (subWorkflowMetadata != null) {
+            extractSecondaryLogFiles(secondaryFileSet,logFileSet,subWorkflowMetadata);
+        }
+    }
+
+    private void extractFilesFromValue(Set<String> fileSet, Object output) {
+        JsonNode node = mapper.valueToTree(output);
```

Comment:
I would suggest moving the conversion to a `JsonNode` into the `extractSecondaryLogFilesFromCall` or `extractSecondaryLogFiles` methods. Having here ends up redundantly calling Jackson to convert an existing JsonNode into another JsonNode

Reply:
Moved the conversion to JsonNode out, conversion takes place when the extractFilesFromValue method is called instead